src/include/duckdb.h

```diff
@@ -701,6 +701,57 @@ typedef void (*duckdb_replacement_callback_t)(duckdb_replacement_scan_info info,
 //===--------------------------------------------------------------------===//
 // Arrow-related types
 //===--------------------------------------------------------------------===//
+struct arrow_schema {
```

Comment:
This is not how this should be done

The definition of these is: `duckdb/src/include/duckdb/common/arrow/arrow.hpp`
The `ARROW_C_DATA_INTERFACE` defines are especially important

And I think putting the arrow API behind a `DUCKDB_ARROW_API` flag also makes sense, so it's opt in
So we're not bringing in all the arrow definitions by default.